### ISSUE TITLE
Hubot suggest doesn't work correctly with HipChat

### ISSUE THREAD

**Issue by SarahKowalik** [hassaku/hubot-suggest#2]
When adding the suggest plugin to our hubot, the following shows up in the logs:

```
[Tue Oct 21 2014 23:22:51 GMT+1100 (EST)] ERROR TypeError: Cannot call method 'match' of undefined
  at /home/hubot/hubot-lore/node_modules/hubot-suggest/scripts/suggest.coffee:66:83, <js>:71:24
  at Listener.callback (/home/hubot/hubot-lore/node_modules/hubot/src/robot.coffee:184:52, <js>:131:16)
  at Listener.call (/home/hubot/hubot-lore/node_modules/hubot/src/listener.coffee:27:7, <js>:23:14)
  at Robot.receive (/home/hubot/hubot-lore/node_modules/hubot/src/robot.coffee:197:9, <js>:142:33)
  at Robot.receive (/home/hubot/hubot-lore/node_modules/hubot/src/robot.coffee:204:7, <js>:153:21)
  at HipChat.Adapter.receive (/home/hubot/hubot-lore/node_modules/hubot/src/adapter.coffee:66:5, <js>:47:25)
  at /home/hubot/hubot-lore/node_modules/hubot-hipchat/src/hipchat.js:245:26
  at invokeCallback (/home/hubot/hubot-lore/node_modules/hubot-hipchat/node_modules/rsvp/node/rsvp.js:161:15)
  at Object.<anonymous> (/home/hubot/hubot-lore/node_modules/hubot-hipchat/node_modules/rsvp/node/rsvp.js:193:9)
  at /home/hubot/hubot-lore/node_modules/hubot-hipchat/node_modules/rsvp/node/rsvp.js:13:16
  at process._tickCallback (node.js:415:13)
```

This should read:

```
module.exports = (robot) ->
  robot.catchAll (msg) ->
    message = msg.message
    showSuggestions(robot.commands, message, robot.adapter, robot.name) if message.match ///^#{robot.name} .*$///i
    msg.finish()
```

After fixing this, hubot doesn't offer suggestions in a normal channel, but does if asked via private message.

When echoing back message with msg.send message, the following is shown:
![screen shot 2014-10-22 at 12 00 08 am](https://cloud.githubusercontent.com/assets/3038925/4718587/46865d1a-5922-11e4-90a8-5c0abf5a62b5.png)

Updating the regex to the following has not made our bot start talking in normal channels:

```
showSuggestions(robot.commands, message, robot.adapter, robot.name) if message.match ///^#{robot.name}: .*$///i
```

**Comment by toadkicker** [comment]
Does 7 & 8 resolve the bug?

**Comment by hassaku** [comment]
I guess that this seems to be solved by #8. 
Please feel free to reopen or create issues if you need.
